### PR TITLE
Fix getting git branch name when status.short = true

### LINKED ISSUE #1822: Prompt `curr_branch` breaks when git config has status.short
If git config contains:
```ini
[status]
    short = true
```
then [the current method of getting the git branch](https://github.com/xonsh/xonsh/blob/619ae11486f2e4febfc2218c0d53c7b3f3c9ea49/xonsh/prompt/vc_branch.py#L16-L25) fails, even if `status.branch = true` is also set.

`git status` is a heavy-handed way of doing this anyway; better would be `git rev-parse --abbrev-ref HEAD` IMO. (PR to follow.)

### PR DESCRIPTION
**Pull request by OJFord**
This PR fixes #1822.

### CODE REVIEW

**Comment by laerus** [comment]
Hey @OJFord! The only problem with `git rev-parse --abbrev-ref HEAD` is that when your HEAD is detached on a commit what you get back is just `HEAD` and not the actually commit hash. I think we can get around #1822 by adding `--long` : `git status --long`.

**Comment by OJFord** [comment]
Ah, true. `--long` works. Another alternative would be `git branch`, and grabbing the line that starts with `* ` - I mention it only because it's probably faster, though I'm not sure how often it would be noticeable in the real-world.

**Comment by laerus** [comment]
For slower machines it would really make a difference cause we call git in each prompt update. Though keep in mind that we can't parse the line using English words because we had problems with ppl having a different default language for git, but it could be done in the same way the current implementation grabs the branch, that is strip/split and get the last 'word'.

**Comment by OJFord** [comment]
@laerus Yep, this has no reliance on specific words, just the `* ` present at the beginning of the current branch.

I have tested this locally with `git config status.short true` on this branch and on master (where the `curr_branch` in prompt becomes "Dockerfile" 😆 ) and with a detached `HEAD`, where this branch also now shows the commit SHA.

**Comment by OJFord** [comment]
I'm not sure what the lint problem is by the way - it doesn't link anything and when I run pylint locally nothing it reports was introduced by my commits.

**Comment by scopatz** [comment]
> I'm not sure what the lint problem is by the way - it doesn't link anything and when I run pylint locally nothing it reports was introduced by my commits.

Thanks for putting this in @OJFord!  This is our own PR-linter, not a code linter.  It says that you need to put in a new entry as per our dev guide here: http://xon.sh/devguide.html#changelog  This is not hard but really helps users.

**Comment by OJFord** [comment]
Ah! Sorry about that, should've read more closely. Fixed.

File: news/git_branch.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+**Changed:** None
+
+**Deprecated:** None
+
+**Removed:** None
+
+**Fixed:**
+
+* `curr_branch` reports correctly when `git config status.short true` is used
```
Comment (scopatz) [inline news/git_branch.rst]: Need to use double backticks in rST.

Comment (OJFord) [inline news/git_branch.rst]: Bah, thanks!

**Comment by scopatz** [comment]
@laerus - I'll leave this to you since I haven't really been following the git status stuff at all :).

**Comment by laerus** [comment]
This works great, thank you @OJFord!

**Pull request merged**